File: benchmarks/pandas/bench_str_normalize.py
"""Benchmark: str normalize on 100k-element string Series"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
data = [f"caf\u00e9 {i}" for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.normalize("NFC")

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.normalize("NFC")
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "str_normalize", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_str_normalize.ts
/**
 * Benchmark: strNormalize on 100k-element string Series
 */
import { Series, strNormalize } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const data = Array.from({ length: ROWS }, (_, i) => `caf\u00e9 ${i}`);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) strNormalize(s, "NFC");
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) strNormalize(s, "NFC");
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "str_normalize",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
